### DISCUSSION TITLE
How to implement logical scrolling (by items)?

### DISCUSSION THREAD

**Discussion by Agarvaen** [AvaloniaUI/Avalonia#12749] [Q&A]
I don't understand how to do logical scrolling. Just basic ListBox (or ItemsControl inside ScrollViewer) with TextBox items. In WPF we have CanContentScroll flag for logical scrolling. No such thing in Avalonia. Found ILogicalScrollable interface but it seems too much and no example found.
Could someone help me? (Examples, direction)

**Comment by wieslawsoltes** [comment]
Some examples:
- https://github.com/wieslawsoltes/HexView/blob/4dd5c91c79fb6f8070e14f7787c6db9f5e9f9e40/HexView/Controls/HexViewControl.cs#L15
- https://github.com/wieslawsoltes/PanAndZoom/blob/7a8cb22eecd8cb6d63f12a1777bf8c453be0a858/src/Avalonia.Controls.PanAndZoom/ZoomBorder.ILogicalScrollable.cs#L11

**Reply by Agarvaen** [reply]
Thank you!  Will take a look.
